zulipterminal/ui_tools/views.py

```diff
@@ -798,4 +798,176 @@ def __init__(self, controller: Any, msg: Any) -> None:
             self.controller.exit_popup()
+        elif is_command_key('EDIT_HISTORY', key):
+            self.controller.show_edit_history(self.msg)
         return super(MsgInfoView, self).keypress(size, key)
+
+
+class EditHistoryView(urwid.ListBox):
+
+    def __init__(self, controller: Any, msg: Any):
+        self.controller = controller
+        self.msg = msg
+
+        self.width = 70
+        self.height = 0  # Calculated depending on contents to show
+        self.to_show = []  # type: List[urwid.Pile]
+
+        if 'edit_history' in self.msg:
+            # Current version
+            timestamp = time.ctime(msg['last_edit_timestamp'])[:-5]
+            topic = [msg['subject']]
+            # tracks topic for versions with no topic change
+            orig_topic = topic
+            if 'prev_subject' in msg['edit_history'][0]:
+                topic.append(" (edited)")
+            content = [msg['content_to_show']]
+            # tracks content for versions with no content change
+            orig_content = content
+
+            # display who edited if edit_author not sender
+            if msg['edit_history'][0]['user_id'] == msg['sender_id']:
+                self.to_show.append(
+                    urwid.Pile([
+                        urwid.Columns([
+                            urwid.AttrWrap(
+                                urwid.Text(topic), 'edit_topic'),
+                            (20, urwid.AttrWrap(
+                                urwid.Text(timestamp, align='right'),
+                                'edit_time'))
+                                ]),
+                        urwid.AttrWrap(
+                            urwid.Text("(Current version)"), 'edit_history'),
+                        urwid.Text([content, "\n"])
+                        ])
+                    )
+            else:
+                # if edit_author not sender, find author by user id
+                edit_author = ''
+                for user in self.controller.view.users:
+                    if user['user_id'] == msg['edit_history'][0]['user_id']:
+                        edit_author = user['full_name']
+                self.to_show.append(
+                    urwid.Pile([
+                        urwid.Columns([
+                            (urwid.AttrWrap(
+                                urwid.Text(topic), 'edit_topic')),
+                            (20, urwid.AttrWrap(
+                                urwid.Text(timestamp, align='right'),
+                                'edit_time'))
+                            ]),
+                        urwid.AttrWrap(
+                            urwid.Text("(Current version)"), 'edit_history'),
+                        urwid.Text([content]),
+                        urwid.AttrWrap(
+                            urwid.Text(["edited by ", edit_author, "\n"],
+                                       align='right'), 'edit_history'),
```

Comment:
Could you simplify the above two `Pile`s using the same `edit_message_history`, at least if you adjust slightly how that method works?